### PR TITLE
[LWIP] Correctly synchronize to lwIP 1.4.1

### PR DESCRIPTION
**Pull request by hpoussin**
## Purpose

There have been numerous attempts to upgrade lwIP library (PR #2001, PR #4999).
However, the current version 1.4.1 is not clearly identified.
Redo an synchronization to version 1.4.1, and document how it has been done

JIRA issue: [CORE-7140](https://jira.reactos.org/browse/CORE-7140)

## Proposed changes

- rename lwip library to lwipcore (lwIP 2.x will introduce a cmake library named lwipcore)
- move ReactOS-code outside of lwip directory
- redo and document how the code is synchronized with lwIP 1.4.1

Changes are mostly in unit test files (not used) and CHANGELOG file.

## TODO

No TODO now.
However, a later PR will be required to upgrade to lwIP 2.x

### CODE REVIEW

File: drivers/network/tcpip/lwip/src/api/api_lib.c [hunk 1]
```diff
@@ -3,12 +3,12 @@
  * Sequential API External module
  *
  */
-
+
```
Comment (SergeGautherie) [inline drivers/network/tcpip/lwip/src/api/api_lib.c:6]: Would it be possible to split all whitespace additions to their own commit?

**Comment by hpoussin** [comment]
Yes, lwIP 2.x will introduce a library named lwipcore in src/Filelists.cmake
```
# This file is indended to be included in end-user CMakeLists.txt
# include(/path/to/Filelists.cmake)
```
We will only have to change `add_subdirectory(lwip)` by `include(${LWIP_DIR}/src/Filelists.cmake)` in drivers/network/tcpip/CMakeLists.txt , when upgrading to lwIP 2.x

You can also think at lwipcore library as code provided by lwIP, while lwip library is glue code provided by ReactOS.

**Review by HBelusca — APPROVED** [review]
OK if it doesn't regress master. Let's get this out so that you can proceed with later lwIP versions syncs.
